sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/DefaultOfferRequirementProvider.java

```diff
@@ -321,6 +322,34 @@ private static void validateTaskRequirements(List<TaskRequirement> taskRequireme
         return resources;
     }
 
+    /**
+     * Returns the ExecutorInfo of a PodInstance if it is still running so it may be re-used, otherwise
+     * it returns a new ExedcutorInfo.
+     * @param podInstance A PodInstance
+     * @return The appropriate ExecutorInfo.
+     */
+    private Protos.ExecutorInfo getExecutor(PodInstance podInstance) {
+        List<Protos.TaskInfo> podTasks = TaskUtils.getPodTasks(podInstance, stateStore);
+
+        List<String> taskNames = podTasks.stream()
+                .map(taskInfo -> taskInfo.getName())
+                .collect(Collectors.toList());
+
+        LOGGER.info("For pod: {}, found tasks: {}", podInstance.getName(), taskNames);
+
+        for (Protos.TaskInfo taskInfo : podTasks) {
+            Optional<Protos.TaskStatus> taskStatusOptional = stateStore.fetchStatus(taskInfo.getName());
+            if (taskStatusOptional.isPresent()
+                    && taskStatusOptional.get().getState() == Protos.TaskState.TASK_RUNNING) {
+                LOGGER.info("Reusing executor: ", taskInfo.getExecutor());
+                return taskInfo.getExecutor();
+            }
+        }
+
+        LOGGER.info("Getting new executor.");
```

Comment:
`"Creating new executor for pod {}, as no RUNNING tasks were found", podInstance.getName()`